### PR TITLE
Flux redux

### PR DESCRIPTION
**Pull request by adrian-ponce-wolox**
## Summary ##
* Model and implement book list, filter, login, logout using redux
* No visual changes

### CODE REVIEW

File: w-books/src/App/screens/Home/index.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 import BookGrid from  '../../components/BookGrid/index.js';
 import FilterSelector from './components/FilterSelector/index.js';
 import FilterInput from './components/FilterInput/index.js';
 import SearchButton from './components/SearchButton/index.js';
-import { getBooks } from '../../../services/BookService.js';
+import { actionCreators as booksActions } from '../../../redux/books/actions.js'
 
 class Home extends Component {
-  state = {filterType: "Author", filterInput: null, filteredList: [], bookList: []}
+  state = {filterType: "Author", filterInput: null}
```
Comment (sbalay) [inline w-books/src/App/screens/Home/index.js]: Declare constants like the filterTypes in a separate file called constants.

File: w-books/src/App/screens/Home/index.js [hunk 2]
```diff
@@ -46,11 +32,23 @@ class Home extends Component {
...
 }
 
-export default Home;
+Home.defaultProps = {
+  bookList: [],
+  filteredList: []
+};
+
+const mapStateToProps = (state) => ({
+  bookList: state.books.list,
```
Comment (sbalay) [inline w-books/src/App/screens/Home/index.js]: this props is not being used

File: w-books/src/App/screens/Home/index.js [hunk 3]
```diff
@@ -46,11 +32,23 @@ class Home extends Component {
...
-          <BookGrid books={this.state.filteredList}/>
+          <BookGrid books={this.props.filteredList}/>
         </div>
       </div>
     );
   }
 }
 
-export default Home;
+Home.defaultProps = {
```
Comment (sbalay) [inline w-books/src/App/screens/Home/index.js:43]: Declare [propTypes](https://facebook.github.io/react/docs/typechecking-with-proptypes.html) too.

File: w-books/src/redux/auth/actions.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { push } from 'react-router-redux'
+import Authenticate, { logOut as logOutService } from '../../services/AuthService.js'
+
+export const actionCreators = {
+  login({email, password}) {
+    return async dispatch => {
+      dispatch({ type: 'LOGIN' });
```
Comment (sbalay) [inline w-books/src/redux/auth/actions.js]: declare actions types as constants too.

File: w-books/src/redux/auth/actions.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      try {
+        Authenticate({email: email, password: password})
+        .then((token) => {
+          dispatch(push('/dashboard'))
```
Comment (sbalay) [inline w-books/src/redux/auth/actions.js]: declare routes as constants too

File: w-books/src/redux/auth/actions.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        });
+      } catch (e) {
+        dispatch(actionCreators.loginFailure(e));
+      }
+    };
+  },
+  loginSuccess(token) {
+    return {
+      type: 'LOGIN_SUCCESS',
+      payload: { token }
```
Comment (sbalay) [inline w-books/src/redux/auth/actions.js]: do we need the token here? we don't really save that value in the store

File: w-books/src/redux/auth/reducer.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    case 'LOGIN':
+      return {
+        ...state,
+        loading: true
+      }
+    case 'LOGIN_SUCCESS':
+      return {
+        ...state,
+        loading: false,
+        token: action.token
```
Comment (sbalay) [inline w-books/src/redux/auth/reducer.js]: Do we need to save the token in the store?

File: w-books/src/redux/book/reducer.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
+const initialState = { loading: false, detail: null, comments: null, suggestions: null }
+
+const book = (state = initialState, action) => {
+  switch (action.type) {
+    case 'FETCH_BOOK':
```
Comment (sbalay) [inline w-books/src/redux/book/reducer.js]: all these case statements look pretty similar. can you think of a way of avoiding repetition?

**Review by sbalay — APPROVED** [review]

File: w-books/src/redux/books/reducer.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        list: action.books,
+        filteredList: action.books
+      }
+    case 'FETCH_BOOKS_FAILURE':
+      return {
+        ...state,
+        loading: false
+      }
+    case 'FILTER_BOOKS':
+      let filteredList = state.list
```
Comment (IgnacioAbadie) [inline w-books/src/redux/books/reducer.js]: is const

**Review by IgnacioAbadie — APPROVED** [review]
Check Seba's Comments and go!

**Pull request merged**